### PR TITLE
make deditec_relais8 thread safe

### PR DESCRIPTION
**Pull request by danielschickmunit**
**Description**

Fixes Bug, when using the deditec_relais_8 in parallel it would crash with `USB Resource Busy`
this is fixed by creating a lock file and checking when getting and setting status if the resource is currently in use by annother labgrid thread or process

I've tested this in my local setup where multiple devices are tested in parallel via a shell script that executes pytest

### CODE REVIEW

File: labgrid/util/agents/deditec_relais8.py [hunk 1]
```diff
@@ -16,6 +16,29 @@
...
+        open(fname, 'a').close()
+
+class Locker:
+    def __init__(self, busnum, devnum):
+        self.busnum = busnum
+        self.devnum = devnum
+
+    def __enter__ (self):
+        touch(f"./{self.busnum}:{self.devnum}.lock")
+        self.fp = open(f"./{self.busnum}:{self.devnum}.lock")
```
[the remark this answers is not in the document]
Comment (Bastian-Krause) [reply labgrid/util/agents/deditec_relais8.py:35]: @danielschickmunit Did you try that?

**Comment by Emantor** [comment]
Hi @danielschickmunit, do you think you will find time for this?

**Comment by ka98** [comment]
hi, I'm @danielschickmunit 

Unfortunatly I do not longer have access to the Github account of my old work and the HW, so I'm not able to continue working on this PR. Sry for the late response.

**Comment by Bastian-Krause** [comment]
@ka98 If you're willing to work on this, simply create a new PR mentioning this. We can then close this.

**Comment by Bastian-Krause** [comment]
If you're still interested in this, we can re-open this or create a new PR. Closing this for now.

**Pull request closed without merging**